s3transfer/futures.py

```diff
@@ -128,8 +128,9 @@ def set_result(self, result):
         Implies that the TransferFuture succeeded.
         """
         with self._lock:
-            self._result = result
-            self._status = 'success'
+            if self._exception is None:
+                self._result = result
+                self._status = 'success'
```

Comment:
Tangentially related, but I noticed you can cancel a task after it's completed.  Is that intentional?  Seems like we should treat cancelled as a terminal state and not allow any transitions out of that state.